diff --git a/tools/llvm-config/llvm-config.cpp b/tools/llvm-config/llvm-config.cpp
index 879b9ab..073b876 100644
--- a/tools/llvm-config/llvm-config.cpp
+++ b/tools/llvm-config/llvm-config.cpp
@@ -349,25 +349,14 @@ int main(int argc, char **argv) {
                              /*IncludeNonInstalled=*/IsInDevelopmentTree);
 
     if (PrintLibs || PrintLibNames || PrintLibFiles) {
-      for (unsigned i = 0, e = RequiredLibs.size(); i != e; ++i) {
-        StringRef Lib = RequiredLibs[i];
-        if (i)
-          OS << ' ';
-
-        if (PrintLibNames) {
-          OS << Lib;
-        } else if (PrintLibFiles) {
-          OS << ActiveLibDir << '/' << Lib;
-        } else if (PrintLibs) {
-          // If this is a typical library name, include it using -l.
-          if (Lib.startswith("lib") && Lib.endswith(".a")) {
-            OS << "-l" << Lib.slice(3, Lib.size()-2);
-            continue;
-          }
-
-          // Otherwise, print the full path.
-          OS << ActiveLibDir << '/' << Lib;
-        }
+      if (PrintLibNames) {
+        OS << "libLLVM.so libLTO.so";
+      } else if (PrintLibFiles) {
+        OS << ActiveLibDir << '/' << "libLLVM.so";
+        OS << " ";
+        OS << ActiveLibDir << '/' << "libLTO.so";
+      } else if (PrintLibs) {
+        OS << "-lLLVM -lLTO";
       }
       OS << '\n';
     }
