From d3d7a9b22c5763865a4334554cd634d89f5f2e0b Mon Sep 17 00:00:00 2001
From: "William A. Kennington III" <wak@google.com>
Date: Tue, 1 Nov 2016 21:04:49 -0700
Subject: [PATCH 1/1] Allow busybox to be in /nix/store/xxxxx-busybox

---
 libbb/appletlib.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libbb/appletlib.c b/libbb/appletlib.c
index 9425c7b..edc3aaa 100644
--- a/libbb/appletlib.c
+++ b/libbb/appletlib.c
@@ -887,7 +887,7 @@ void FAST_FUNC run_applet_no_and_exit(int applet_no, char **argv)
 static NORETURN void run_applet_and_exit(const char *name, char **argv)
 {
 #  if ENABLE_BUSYBOX
-	if (is_prefixed_with(name, "busybox"))
+	if (strstr(name, "busybox") != NULL)
 		exit(busybox_main(argv));
 #  endif
 #  if NUM_APPLETS > 0
-- 
2.8.0.rc3.226.g39d4020

