From 57c0558390cdf33ffcac067c4b6faf737db4e951 Mon Sep 17 00:00:00 2001
From: "William A. Kennington III" <william@wkennington.com>
Date: Tue, 13 Mar 2018 15:08:07 -0700
Subject: [PATCH 1/4] libcpp: Determine if we should enforce purity

---
 libcpp/macro.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libcpp/macro.c b/libcpp/macro.c
index 776af7bd00e..5d6b8aa11d2 100644
--- a/libcpp/macro.c
+++ b/libcpp/macro.c
@@ -386,6 +386,7 @@ _cpp_builtin_macro_text (cpp_reader *pfile, cpp_hashnode *node,
 {
   const uchar *result = NULL;
   linenum_type number = 1;
+  bool enforce_purity = getenv("NIX_BUILD_TOP") != NULL;
 
   switch (node->value.builtin)
     {
-- 
2.18.0

