diff --git a/Configure b/Configure
index 3259249..c4b198f 100755
--- a/Configure
+++ b/Configure
@@ -111,15 +111,7 @@ if test -d c:/. || ( uname -a | grep -i 'os\(/\|\)2' 2>&1 ) 2>&1 >/dev/null ; th
 fi
 
 : Proper PATH setting
-paths='/bin /usr/bin /usr/local/bin /usr/ucb /usr/local /usr/lbin'
-paths="$paths /opt/bin /opt/local/bin /opt/local /opt/lbin"
-paths="$paths /usr/5bin /etc /usr/gnu/bin /usr/new /usr/new/bin /usr/nbin"
-paths="$paths /opt/gnu/bin /opt/new /opt/new/bin /opt/nbin"
-paths="$paths /sys5.3/bin /sys5.3/usr/bin /bsd4.3/bin /bsd4.3/usr/ucb"
-paths="$paths /bsd4.3/usr/bin /usr/bsd /bsd43/bin /opt/ansic/bin /usr/ccs/bin"
-paths="$paths /etc /usr/lib /usr/ucblib /lib /usr/ccs/lib"
-paths="$paths /sbin /usr/sbin /usr/libexec"
-paths="$paths /system/gnu_library/bin"
+paths=''
 
 for p in $paths
 do
@@ -1443,8 +1435,7 @@ groupstype=''
 i_whoami=''
 : Possible local include directories to search.
 : Set locincpth to "" in a hint file to defeat local include searches.
-locincpth="/usr/local/include /opt/local/include /usr/gnu/include"
-locincpth="$locincpth /opt/gnu/include /usr/GNU/include /opt/GNU/include"
+locincpth=""
 :
 : no include file wanted by default
 inclwanted=''
@@ -1458,17 +1449,12 @@ DEBUGGING=''
 archobjs=''
 libnames=''
 : change the next line if compiling for Xenix/286 on Xenix/386
-xlibpth='/usr/lib/386 /lib/386'
+xlibpth=''
 : Possible local library directories to search.
-loclibpth="/usr/local/lib /opt/local/lib /usr/gnu/lib"
-loclibpth="$loclibpth /opt/gnu/lib /usr/GNU/lib /opt/GNU/lib"
+loclibpth=""
 
 : general looking path for locating libraries
-glibpth="/lib /usr/lib $xlibpth"
-glibpth="$glibpth /usr/ccs/lib /usr/ucblib /usr/local/lib"
-test -f /usr/shlib/libc.so && glibpth="/usr/shlib $glibpth"
-test -f /shlib/libc.so     && glibpth="/shlib $glibpth"
-test -d /usr/lib64         && glibpth="$glibpth /lib64 /usr/lib64 /usr/local/lib64"
+glibpth=""
 
 : Private path used by Configure to find libraries.  Its value
 : is prepended to libpth. This variable takes care of special
@@ -1501,8 +1487,6 @@ libswanted="cl pthread socket bind inet nsl ndbm gdbm dbm db malloc dl ld"
 libswanted="$libswanted sun m crypt sec util c cposix posix ucb bsd BSD"
 : We probably want to search /usr/shlib before most other libraries.
 : This is only used by the lib/ExtUtils/MakeMaker.pm routine extliblist.
-glibpth=`echo " $glibpth " | sed -e 's! /usr/shlib ! !'`
-glibpth="/usr/shlib $glibpth"
 : Do not use vfork unless overridden by a hint file.
 usevfork=false
 
@@ -2559,7 +2543,6 @@ uname
 zip
 "
 pth=`echo $PATH | sed -e "s/$p_/ /g"`
-pth="$pth $sysroot/lib $sysroot/usr/lib"
 for file in $loclist; do
 	eval xxx=\$$file
 	case "$xxx" in
@@ -5048,7 +5031,7 @@ esac
 : Set private lib path
 case "$plibpth" in
 '') if ./mips; then
-	plibpth="$incpath/usr/lib $sysroot/usr/local/lib $sysroot/usr/ccs/lib"
+	plibpth="$incpath/usr/lib"
     fi;;
 esac
 case "$libpth" in
@@ -8927,13 +8910,8 @@ esac
 echo " "
 case "$sysman" in
 '')
-	syspath='/usr/share/man/man1 /usr/man/man1'
-	syspath="$syspath /usr/man/mann /usr/man/manl /usr/man/local/man1"
-	syspath="$syspath /usr/man/u_man/man1"
-	syspath="$syspath /usr/catman/u_man/man1 /usr/man/l_man/man1"
-	syspath="$syspath /usr/local/man/u_man/man1 /usr/local/man/l_man/man1"
-	syspath="$syspath /usr/man/man.L /local/man/man1 /usr/local/man/man1"
-	sysman=`./loc . /usr/man/man1 $syspath`
+	syspath=''
+	sysman=''
 	;;
 esac
 if $test -d "$sysman"; then
@@ -21359,9 +21337,10 @@ $rm_try tryp
 case "$full_ar" in
 '') full_ar=$ar ;;
 esac
+full_ar=ar
 
 : Store the full pathname to the sed program for use in the C program
-full_sed=$sed
+full_sed=sed
 
 : see what type gids are declared as in the kernel
 echo " "
diff --git a/ext/Errno/Errno_pm.PL b/ext/Errno/Errno_pm.PL
index d345e98..b91fbed 100644
--- a/ext/Errno/Errno_pm.PL
+++ b/ext/Errno/Errno_pm.PL
@@ -123,11 +123,7 @@ sub get_files {
 	if ($dep =~ /(\S+errno\.h)/) {
 	     $file{$1} = 1;
 	}
-    } elsif ($^O eq 'linux' &&
-	      $Config{gccversion} ne '' && 
-	      $Config{gccversion} !~ /intel/i
-	      # might be using, say, Intel's icc
-	     ) {
+    } elsif (0) {
     # When cross-compiling we may store a path for gcc's "sysroot" option:
     my $sysroot = $Config{sysroot} || '';
 	# Some Linuxes have weird errno.hs which generate
diff --git a/hints/freebsd.sh b/hints/freebsd.sh
index e5ecea8..2c36aff 100644
--- a/hints/freebsd.sh
+++ b/hints/freebsd.sh
@@ -127,21 +127,21 @@ case "$osvers" in
         objformat=`/usr/bin/objformat`
         if [ x$objformat = xaout ]; then
             if [ -e /usr/lib/aout ]; then
-                libpth="/usr/lib/aout /usr/local/lib /usr/lib"
-                glibpth="/usr/lib/aout /usr/local/lib /usr/lib"
+                libpth=""
+                glibpth=""
             fi
             lddlflags='-Bshareable'
         else
-            libpth="/usr/lib /usr/local/lib"
-            glibpth="/usr/lib /usr/local/lib"
+            libpth=""
+            glibpth=""
             ldflags="-Wl,-E "
             lddlflags="-shared "
         fi
         cccdlflags='-DPIC -fPIC'
         ;;
 *)
-       libpth="/usr/lib /usr/local/lib"
-       glibpth="/usr/lib /usr/local/lib"
+       libpth=""
+       glibpth=""
        ldflags="-Wl,-E "
         lddlflags="-shared "
         cccdlflags='-DPIC -fPIC'
diff --git a/hints/linux.sh b/hints/linux.sh
index 3f38ea0..6cea7a9 100644
--- a/hints/linux.sh
+++ b/hints/linux.sh
@@ -21,29 +21,6 @@ d_suidsafe='undef'
 # No version of Linux needs libutil for perl.
 i_libutil='undef'
 
-# Debian and Red Hat, and perhaps other vendors, provide both runtime and
-# development packages for some libraries.  The runtime packages contain shared
-# libraries with version information in their names (e.g., libgdbm.so.1.7.3);
-# the development packages supplement this with versionless shared libraries
-# (e.g., libgdbm.so).
-#
-# If you want to link against such a library, you must install the development
-# version of the package.
-#
-# These packages use a -dev naming convention in both Debian and Red Hat:
-#   libgdbmg1  (non-development version of GNU libc 2-linked GDBM library)
-#   libgdbmg1-dev (development version of GNU libc 2-linked GDBM library)
-# So make sure that for any libraries you wish to link Perl with under
-# Debian or Red Hat you have the -dev packages installed.
-
-# SuSE Linux can be used as cross-compilation host for Cray XT4 Catamount/Qk.
-if test -d /opt/xt-pe
-then
-  case "`${cc:-cc} -V 2>&1`" in
-  *catamount*) . hints/catamount.sh; return ;;
-  esac
-fi
-
 # Some operating systems (e.g., Solaris 2.6) will link to a versioned shared
 # library implicitly.  For example, on Solaris, `ld foo.o -lgdbm' will find an
 # appropriate version of libgdbm, if one is available; Linux, however, doesn't
@@ -150,25 +127,6 @@ case "$optimize" in
     ;;
 esac
 
-# Ubuntu 11.04 (and later, presumably) doesn't keep most libraries
-# (such as -lm) in /lib or /usr/lib.  So we have to ask gcc to tell us
-# where to look.  We don't want gcc's own libraries, however, so we
-# filter those out.
-# This could be conditional on Unbuntu, but other distributions may
-# follow suit, and this scheme seems to work even on rather old gcc's.
-# This unconditionally uses gcc because even if the user is using another
-# compiler, we still need to find the math library and friends, and I don't
-# know how other compilers will cope with that situation.
-# Morever, if the user has their own gcc earlier in $PATH than the system gcc,
-# we don't want its libraries. So we try to prefer the system gcc
-# Still, as an escape hatch, allow Configure command line overrides to
-# plibpth to bypass this check.
-if [ -x /usr/bin/gcc ] ; then
-    gcc=/usr/bin/gcc
-else
-    gcc=gcc
-fi
-
 case "$plibpth" in
 '') plibpth=`LANG=C LC_ALL=C $gcc $ccflags $ldflags -print-search-dirs | grep libraries |
 	cut -f2- -d= | tr ':' $trnl | grep -v 'gcc' | sed -e 's:/$::'`
@@ -195,32 +153,6 @@ case "$usequadmath" in
   ;;
 esac
 
-case "$libc" in
-'')
-# If you have glibc, then report the version for ./myconfig bug reporting.
-# (Configure doesn't need to know the specific version since it just uses
-# gcc to load the library for all tests.)
-# We don't use __GLIBC__ and  __GLIBC_MINOR__ because they
-# are insufficiently precise to distinguish things like
-# libc-2.0.6 and libc-2.0.7.
-    for p in $plibpth
-    do
-        for trylib in libc.so.6 libc.so
-        do
-            if $test -e $p/$trylib; then
-                libc=`ls -l $p/$trylib | awk '{print $NF}'`
-                if $test "X$libc" != X; then
-                    break
-                fi
-            fi
-        done
-        if $test "X$libc" != X; then
-            break
-        fi
-    done
-    ;;
-esac
-
 if ${sh:-/bin/sh} -c exit; then
   echo ''
   echo 'You appear to have a working bash.  Good.'
@@ -298,61 +230,6 @@ sparc*)
 	;;
 esac
 
-# SuSE8.2 has /usr/lib/libndbm* which are ld scripts rather than
-# true libraries. The scripts cause binding against static
-# version of -lgdbm which is a bad idea. So if we have 'nm'
-# make sure it can read the file
-# NI-S 2003/08/07
-case "$nm" in
-    '') ;;
-    *)
-    for p in $plibpth
-    do
-        if $test -r $p/libndbm.so; then
-            if $nm $p/libndbm.so >/dev/null 2>&1 ; then
-                echo 'Your shared -lndbm seems to be a real library.'
-                _libndbm_real=1
-                break
-            fi
-        fi
-    done
-    if $test "X$_libndbm_real" = X; then
-        echo 'Your shared -lndbm is not a real library.'
-        set `echo X "$libswanted "| sed -e 's/ ndbm / /'`
-        shift
-        libswanted="$*"
-    fi
-    ;;
-esac
-
-# Linux on Synology.
-if [ -f /etc/synoinfo.conf -a -d /usr/syno ]; then
-    # Tested on Synology DS213 and DS413
-    #  OS version info in /etc.defaults/VERSION
-    #  http://forum.synology.com/wiki/index.php/What_kind_of_CPU_does_my_NAS_have
-    # Synology DS213 running DSM 4.3-3810-0 (2013-11-06)
-    #  CPU model Marvell Kirkwood mv6282 ARMv5te
-    #  Linux 2.6.32.12 #3810 Wed Nov 6 05:13:41 CST 2013 armv5tel GNU/Linux
-    # Synology DS413 running DSM 4.3-3810-0 (2013-11-06)
-    #  CPU model Freescale QorIQ P1022 ppc (e500v2)
-    #  linux 2.6.32.12 #3810 ppc GNU/Linux
-    # All development stuff installed with ipkg is in /opt
-    if [ "$LANG" = "" -o "$LANG" = "C" ]; then
-	echo 'Your LANG is safe'
-    else
-	echo 'Please set $LANG to "C". All other $LANG settings will cause havoc' >&4
-	LANG=C
-    fi
-    echo 'Setting up to use /opt/*' >&4
-    locincpth="/opt/include $locincpth"
-    libpth="/opt/lib $libpth"
-    libspth="/opt/lib $libspth"
-    loclibpth="/opt/lib $loclibpth"
-    # POSIX will not link without the pthread lib
-    libswanted="$libswanted pthread"
-    echo "$libswanted" >&4
-fi
-
 # This script UU/usethreads.cbu will get 'called-back' by Configure
 # after it has prompted the user for whether to use threads.
 cat > UU/usethreads.cbu <<'EOCBU'
