diff --git a/src/go/build/gc.go b/src/go/build/gc.go
index e2be2cbb1d..fb7001d601 100644
--- a/src/go/build/gc.go
+++ b/src/go/build/gc.go
@@ -17,6 +17,10 @@ import (
 
 // getToolDir returns the default value of ToolDir.
 func getToolDir() string {
+	s := os.Getenv("GOTOOLDIR")
+	if s != "" {
+		return s
+	}
 	return filepath.Join(runtime.GOROOT(), "pkg/tool/"+runtime.GOOS+"_"+runtime.GOARCH)
 }
 
