diff --git a/nss/nsswitch.c b/nss/nsswitch.c
index bb644cb..41e5ac0 100644
--- a/nss/nsswitch.c
+++ b/nss/nsswitch.c
@@ -355,8 +355,13 @@ nss_load_library (service_user *ni)
 			  ".so"),
 		__nss_shlib_revision);
 
+      const char *abs_path = "/run/current-system/nss/lib/";
+      char abs_shlib_name[strlen(abs_path) + shlen];
+      __stpcpy(__stpcpy(abs_shlib_name, abs_path), shlib_name);
       ni->library->lib_handle = __libc_dlopen (shlib_name);
       if (ni->library->lib_handle == NULL)
+	ni->library->lib_handle = __libc_dlopen (abs_shlib_name);
+      if (ni->library->lib_handle == NULL)
 	{
 	  /* Failed to load the library.  */
 	  ni->library->lib_handle = (void *) -1l;
