diff --git a/src/cargo/core/compiler/build_context/target_info.rs b/src/cargo/core/compiler/build_context/target_info.rs
index 80f1d51c3..255f7342c 100644
--- a/src/cargo/core/compiler/build_context/target_info.rs
+++ b/src/cargo/core/compiler/build_context/target_info.rs
@@ -420,6 +420,16 @@ fn env_args(
     // same as the host, build scripts in plugins won't get
     // RUSTFLAGS.
     if !requested_kind.is_host() && kind.is_host() {
+        // First try RUSTFLAGS_HOST from the environment
+        if let Ok(a) = env::var(format!("{}_HOST", name)) {
+            let args = a
+                .split(' ')
+                .map(str::trim)
+                .filter(|s| !s.is_empty())
+                .map(str::to_string);
+            return Ok(args.collect());
+        }
+
         // This is probably a build script or plugin and we're
         // compiling with --target. In this scenario there are
         // no rustflags we can apply.
