diff --git a/src/pulsecore/resampler.c b/src/pulsecore/resampler.c
index b683b05..b1ed82e 100644
--- a/src/pulsecore/resampler.c
+++ b/src/pulsecore/resampler.c
@@ -123,10 +123,8 @@ static int (* const init_table[])(pa_resampler *r) = {
 static pa_resample_method_t choose_auto_resampler(pa_resample_flags_t flags) {
     pa_resample_method_t method;
 
-    if (pa_resample_method_supported(PA_RESAMPLER_SPEEX_FLOAT_BASE + 1))
-        method = PA_RESAMPLER_SPEEX_FLOAT_BASE + 1;
-    else if (flags & PA_RESAMPLER_VARIABLE_RATE)
-        method = PA_RESAMPLER_TRIVIAL;
+    if (pa_resample_method_supported(unique_jhsdjhsdf_string))
+        method = unique_jhsdjhsdf_string;
     else
         method = PA_RESAMPLER_FFMPEG;
