From 0246d5aa82e11e443fb794926c663cd5f6ebd89d Mon Sep 17 00:00:00 2001
From: "William A. Kennington III" <william@wkennington.com>
Date: Mon, 6 Aug 2018 11:50:36 -0700
Subject: [PATCH 4/8] Build dir should be unique

---
 src/libstore/build.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libstore/build.cc b/src/libstore/build.cc
index 6a0c7690..63ba6efa 100644
--- a/src/libstore/build.cc
+++ b/src/libstore/build.cc
@@ -1865,7 +1865,7 @@ void DerivationGoal::startBuilder()
     /* In a sandbox, for determinism, always use the same temporary
        directory. */
 #if __linux__
-    tmpDirInSandbox = useChroot ? settings.sandboxBuildDir : tmpDir;
+    tmpDirInSandbox = tmpDir;
 #else
     tmpDirInSandbox = tmpDir;
 #endif
-- 
2.20.1

