From 2dd1d5cc97c8afe72365153e929b0cfa6e49624e Mon Sep 17 00:00:00 2001
From: "William A. Kennington III" <william@wkennington.com>
Date: Mon, 6 Aug 2018 14:52:56 -0700
Subject: [PATCH 5/8] Always verify TLS

---
 src/libstore/builtins/fetchurl.cc | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/libstore/builtins/fetchurl.cc b/src/libstore/builtins/fetchurl.cc
index 92aec63a..454ea56c 100644
--- a/src/libstore/builtins/fetchurl.cc
+++ b/src/libstore/builtins/fetchurl.cc
@@ -33,11 +33,9 @@ void builtinFetchurl(const BasicDerivation & drv, const std::string & netrcData)
     auto fetch = [&](const std::string & url) {
 
         auto source = sinkToSource([&](Sink & sink) {
-
-            /* No need to do TLS verification, because we check the hash of
-               the result anyway. */
+            /* Always verify TLS in case the output hash is copied */
             DownloadRequest request(url);
-            request.verifyTLS = false;
+            request.verifyTLS = true;
             request.decompress = false;
 
             auto decompressor = makeDecompressionSink(
-- 
2.20.1

