From f3241eed0498574ad313470ef11b5f8caa90c338 Mon Sep 17 00:00:00 2001
From: "William A. Kennington III" <william@wkennington.com>
Date: Mon, 6 Aug 2018 15:44:03 -0700
Subject: [PATCH 08/10] Make fixed-ouput message nicer

---
 src/libstore/build.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libstore/build.cc b/src/libstore/build.cc
index 365b15ea..d36ec571 100644
--- a/src/libstore/build.cc
+++ b/src/libstore/build.cc
@@ -3138,7 +3138,7 @@ void DerivationGoal::registerOutputs()
                 /* Throw an error after registering the path as
                    valid. */
                 delayedException = std::make_exception_ptr(
-                    BuildError("fixed-output derivation produced path '%s' with %s hash '%s' instead of the expected hash '%s'",
+                    BuildError("fixed-output derivation produced path '%s' with invalid hash(%s):\n  Got: %s\n  Expected: %s",
                         dest, printHashType(h.type), printHash16or32(h2), printHash16or32(h)));
 
                 Path actualDest = worker.store.toRealPath(dest);
-- 
2.19.0.605.g01d371f741-goog

