From 836c3aa9ad9736a5a86999923fefd018cb4cc36b Mon Sep 17 00:00:00 2001
From: "William A. Kennington III" <william@wkennington.com>
Date: Fri, 12 Jul 2019 10:57:06 -0700
Subject: [PATCH 11/11] build: Support optional chroot builds

---
 src/libstore/build.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/libstore/build.cc b/src/libstore/build.cc
index 01a85124..04a21ed0 100644
--- a/src/libstore/build.cc
+++ b/src/libstore/build.cc
@@ -1915,6 +1915,7 @@ void DerivationGoal::startBuilder()
     /* Are we doing a chroot build? */
     {
         auto noChroot = parsedDrv->getBoolAttr("__noChroot");
+        auto optionalChroot = parsedDrv->getBoolAttr("__optionalChroot");
         if (settings.sandboxMode == smEnabled) {
             if (noChroot)
                 throw Error(format("derivation '%1%' has '__noChroot' set, "
@@ -1929,7 +1930,7 @@ void DerivationGoal::startBuilder()
         else if (settings.sandboxMode == smDisabled)
             useChroot = false;
         else if (settings.sandboxMode == smRelaxed)
-            useChroot = !fixedOutput && !noChroot;
+            useChroot = !fixedOutput && !noChroot && !optionalChroot;
     }
 
     if (worker.store.storeDir != worker.store.realStoreDir) {
-- 
2.24.0.432.g9d3f5f5b63-goog

