From be717fd737bb463d9764b586237a18f26df2fcd4 Mon Sep 17 00:00:00 2001
From: "William A. Kennington III" <william@wkennington.com>
Date: Tue, 7 May 2019 20:13:54 -0700
Subject: [PATCH] jmemecust: Fix for new libjpeg

---
 source/fitz/jmemcust.c | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/source/fitz/jmemcust.c b/source/fitz/jmemcust.c
index d0cb3048..73be2123 100644
--- a/source/fitz/jmemcust.c
+++ b/source/fitz/jmemcust.c
@@ -16,7 +16,8 @@
 
 #if !defined(SHARE_JPEG) || SHARE_JPEG==0
 
-#include "jinclude.h"
+#include <stddef.h>
+#include <stdio.h>
 #include "jpeglib.h"
 #include "jmemsys.h"
 #include "jerror.h"
@@ -66,12 +67,12 @@ jpeg_free_large (j_common_ptr cinfo, void FAR * object, size_t sizeofobject)
  * Here we always say, "we got all you want bud!"
  */
 
-GLOBAL(long)
-jpeg_mem_available (j_common_ptr cinfo, long min_bytes_needed,
-		long max_bytes_needed, long already_allocated)
+GLOBAL(size_t)
+jpeg_mem_available (j_common_ptr cinfo, size_t min_bytes_needed,
+		size_t max_bytes_needed, size_t already_allocated)
 {
 	jpeg_cust_mem_data *cmem = GET_CUST_MEM_DATA(cinfo);
-	long ret = max_bytes_needed;
+	size_t ret = max_bytes_needed;
 
 	if (cmem->j_mem_avail)
 		ret = (cmem->j_mem_avail)(cinfo);
-- 
2.21.0.1020.gf2820cf01a-goog

