From 07585b4d5cafbe9ac6bb397a2c99ac0284d8d2d8 Mon Sep 17 00:00:00 2001
From: "William A. Kennington III" <william@wkennington.com>
Date: Sat, 6 Jul 2019 18:08:16 -0700
Subject: [PATCH] Fix paths

---
 include/pathnames.h | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/include/pathnames.h b/include/pathnames.h
index ee470ee4b..1646ca5d9 100644
--- a/include/pathnames.h
+++ b/include/pathnames.h
@@ -21,17 +21,17 @@
 #undef _PATH_DEFPATH
 
 #ifdef USE_USRDIR_PATHS_ONLY
-# define _PATH_DEFPATH	        "/usr/local/bin:/usr/bin"
+# define _PATH_DEFPATH	        "/run/current-system/sw/bin:/usr/local/bin:/usr/bin"
 #else
-# define _PATH_DEFPATH	        "/usr/local/bin:/bin:/usr/bin"
+# define _PATH_DEFPATH	        "/run/current-system/sw/bin:/usr/local/bin:/bin:/usr/bin"
 #endif
 
 #undef _PATH_DEFPATH_ROOT
 
 #ifdef USE_USRDIR_PATHS_ONLY
-# define _PATH_DEFPATH_ROOT	"/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin"
+# define _PATH_DEFPATH_ROOT	"/run/current-system/sw/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin"
 #else
-# define _PATH_DEFPATH_ROOT	"/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin"
+# define _PATH_DEFPATH_ROOT	"/run/current-system/sw/bin:/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin"
 #endif
 
 #define	_PATH_HUSHLOGIN		".hushlogin"
@@ -42,17 +42,17 @@
 #ifndef _PATH_MAILDIR
 # define _PATH_MAILDIR		"/var/spool/mail"
 #endif
-#define	_PATH_MOTDFILE		"/usr/share/misc/motd:/run/motd:/etc/motd"
+#define	_PATH_MOTDFILE		"/run/current-system/sw/share/misc/motd:/usr/share/misc/motd:/run/motd:/etc/motd"
 #ifndef _PATH_NOLOGIN
 # define _PATH_NOLOGIN		"/etc/nologin"
 #endif
 #define	_PATH_VAR_NOLOGIN	"/var/run/nologin"
 
 #ifndef _PATH_LOGIN
-# define _PATH_LOGIN		"/bin/login"
+# define _PATH_LOGIN		"/run/current-system/sw/bin/login"
 #endif
-#define _PATH_SHUTDOWN		"/sbin/shutdown"
-#define _PATH_POWEROFF		"/sbin/poweroff"
+#define _PATH_SHUTDOWN		"/run/current-system/sw/bin/shutdown"
+#define _PATH_POWEROFF		"/run/current-system/sw/bin/poweroff"
 
 #define _PATH_TERMCOLORS_DIRNAME "terminal-colors.d"
 #define _PATH_TERMCOLORS_DIR	"/etc/" _PATH_TERMCOLORS_DIRNAME
@@ -72,13 +72,13 @@
 #define _PATH_ISSUEDIR		_PATH_ISSUE ".d"
 
 #define _PATH_OS_RELEASE_ETC	"/etc/os-release"
-#define _PATH_OS_RELEASE_USR	"/usr/lib/os-release"
+#define _PATH_OS_RELEASE_USR	"/run/current-system/etc/os-release"
 #define _PATH_NUMLOCK_ON	_PATH_RUNSTATEDIR "/numlock-on"
 #define _PATH_LOGINDEFS		"/etc/login.defs"
 
 /* misc paths */
-#define _PATH_WORDS             "/usr/share/dict/words"
-#define _PATH_WORDS_ALT         "/usr/share/dict/web2"
+#define _PATH_WORDS             "/run/current-system/sw/share/dict/words"
+#define _PATH_WORDS_ALT         "/run/current-system/sw/share/dict/web2"
 
 /* mount paths */
 #define _PATH_FILESYSTEMS	"/etc/filesystems"
-- 
2.21.0

