diff --git a/include/pathnames.h b/include/pathnames.h
index 05fb082..5389363 100644
--- a/include/pathnames.h
+++ b/include/pathnames.h
@@ -22,17 +22,17 @@
 #undef _PATH_DEFPATH
 
 #ifdef USE_USRDIR_PATHS_ONLY
-# define _PATH_DEFPATH	        "/usr/local/bin:/usr/bin"
+# define _PATH_DEFPATH	        "/run/current-system/sw/bin:/usr/local/bin:/usr/bin"
 #else
-# define _PATH_DEFPATH	        "/usr/local/bin:/bin:/usr/bin"
+# define _PATH_DEFPATH	        "/run/current-system/sw/bin:/usr/local/bin:/bin:/usr/bin"
 #endif
 
 #undef _PATH_DEFPATH_ROOT
 
 #ifdef USE_USRDIR_PATHS_ONLY
-# define _PATH_DEFPATH_ROOT	"/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin"
+# define _PATH_DEFPATH_ROOT	"/run/current-system/sw/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin"
 #else
-# define _PATH_DEFPATH_ROOT	"/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin"
+# define _PATH_DEFPATH_ROOT	"/run/current-system/sw/bin:/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin"
 #endif
 
 #define _PATH_WTMPLOCK		"/etc/wtmplock"
@@ -52,9 +52,9 @@
 #define	_PATH_VAR_NOLOGIN	"/var/run/nologin"
 
 #ifndef _PATH_LOGIN
-#define _PATH_LOGIN		"/bin/login"
+#define _PATH_LOGIN		"/run/current-system/sw/bin/login"
 #endif
-#define _PATH_SHUTDOWN		"/sbin/shutdown"
+#define _PATH_SHUTDOWN		"/run/current-system/sw/bin/shutdown"
 
 #define _PATH_TERMCOLORS_DIRNAME "terminal-colors.d"
 #define _PATH_TERMCOLORS_DIR	"/etc/" _PATH_TERMCOLORS_DIRNAME
@@ -75,15 +75,15 @@
 /* used in term-utils/agetty.c */
 #define _PATH_ISSUE		"/etc/issue"
 #define _PATH_OS_RELEASE_ETC	"/etc/os-release"
-#define _PATH_OS_RELEASE_USR	"/usr/lib/os-release"
+#define _PATH_OS_RELEASE_USR	"/run/current-system/sw/lib/os-release"
 
 #define _PATH_NUMLOCK_ON	_PATH_LOCALSTATEDIR "/numlock-on"
 
 #define _PATH_LOGINDEFS		"/etc/login.defs"
 
 /* used in misc-utils/look.c */
-#define _PATH_WORDS             "/usr/share/dict/words"
-#define _PATH_WORDS_ALT         "/usr/share/dict/web2"
+#define _PATH_WORDS             "/run/current-system/sw/share/dict/words"
+#define _PATH_WORDS_ALT         "/run/current-system/sw/share/dict/web2"
 
 /* mount paths */
 #define _PATH_FILESYSTEMS	"/etc/filesystems"
