diff --git a/read.c b/read.c
index b870aa8..7bf0dfc 100644
--- a/read.c
+++ b/read.c
@@ -108,10 +108,12 @@ static const char *default_include_directories[] =
 #endif
     INCLUDEDIR,
 #ifndef _AMIGA
+#if 0
     "/usr/gnu/include",
     "/usr/local/include",
     "/usr/include",
 #endif
+#endif
     0
   };
 
diff --git a/remake.c b/remake.c
index df1a9e0..c58d1cf 100644
--- a/remake.c
+++ b/remake.c
@@ -1585,9 +1585,11 @@ library_search (const char *lib, FILE_TIMESTAMP *mtime_ptr)
   static const char *dirs[] =
     {
 #ifndef _AMIGA
+#if 0
       "/lib",
       "/usr/lib",
 #endif
+#endif
 #if defined(WINDOWS32) && !defined(LIBDIR)
 /*
  * This is completely up to the user at product install time. Just define
