diff --git a/src/read.c b/src/read.c
index fa197fb..43d229b 100644
--- a/src/read.c
+++ b/src/read.c
@@ -102,6 +102,7 @@ static struct conditionals *conditionals = &toplevel_conditionals;
 
 static const char *default_include_directories[] =
   {
+#if 0
 #if defined(WINDOWS32) && !defined(INCLUDEDIR)
 /* This completely up to the user when they install MSVC or other packages.
    This is defined as a placeholder.  */
@@ -112,6 +113,7 @@ static const char *default_include_directories[] =
     "/usr/gnu/include",
     "/usr/local/include",
     "/usr/include",
+#endif
 #endif
     0
   };
diff --git a/src/remake.c b/src/remake.c
index fb237c5..59c485b 100644
--- a/src/remake.c
+++ b/src/remake.c
@@ -1600,6 +1600,7 @@ library_search (const char *lib, FILE_TIMESTAMP *mtime_ptr)
 {
   static const char *dirs[] =
     {
+#if 0
 #ifndef _AMIGA
       "/lib",
       "/usr/lib",
@@ -1612,6 +1613,7 @@ library_search (const char *lib, FILE_TIMESTAMP *mtime_ptr)
 #define LIBDIR "."
 #endif
       LIBDIR,                   /* Defined by configuration.  */
+#endif
       0
     };
 
