diff --git a/common/user-list.c b/common/user-list.c
index 58cb028..c391e98 100644
--- a/common/user-list.c
+++ b/common/user-list.c
@@ -353,7 +353,7 @@ load_passwd_file (CommonUserList *user_list, gboolean emit_add_signal)
 
     value = g_key_file_get_string (config, "UserList", "hidden-shells", NULL);
     if (!value)
-        value = g_strdup ("/bin/false /usr/sbin/nologin");
+        value = g_strdup ("/bin/false /usr/sbin/nologin /run/current-system/sw/bin/nologin");
     hidden_shells = g_strsplit (value, " ", -1);
     g_free (value);
 
diff --git a/src/session-child.c b/src/session-child.c
index 28986d3..fcc297e 100644
--- a/src/session-child.c
+++ b/src/session-child.c
@@ -449,7 +449,7 @@ session_child_run (int argc, char **argv)
         else
         {
             /* Set POSIX variables */
-            pam_putenv (pam_handle, "PATH=/usr/local/bin:/usr/bin:/bin");
+            pam_putenv (pam_handle, "PATH=/run/current-system/sw/bin");
             pam_putenv (pam_handle, g_strdup_printf ("USER=%s", username));
             pam_putenv (pam_handle, g_strdup_printf ("LOGNAME=%s", username));
             pam_putenv (pam_handle, g_strdup_printf ("HOME=%s", user_get_home_directory (user)));
diff --git a/src/shared-data-manager.c b/src/shared-data-manager.c
index fd7427e..84764c5 100644
--- a/src/shared-data-manager.c
+++ b/src/shared-data-manager.c
@@ -64,7 +64,7 @@ delete_unused_user (gpointer key, gpointer value, gpointer user_data)
 
     gchar *path = g_build_filename (USERS_DIR, user, NULL);
     gchar *quoted_path = g_shell_quote (path);
-    gchar *cmd = g_strdup_printf ("/bin/rm -rf %s", quoted_path);
+    gchar *cmd = g_strdup_printf ("/run/current-system/sw/bin/rm -rf %s", quoted_path);
 
     g_spawn_command_line_async (cmd, &error);
     if (error)
