Index: trunk/Source/JavaScriptCore/runtime/Options.cpp
===================================================================
--- trunk/Source/JavaScriptCore/runtime/Options.cpp	(revision 194376)
+++ trunk/Source/JavaScriptCore/runtime/Options.cpp	(revision 194518)
@@ -654,5 +654,5 @@
         return m_entry.unsignedVal == other.m_entry.unsignedVal;
     case Options::Type::doubleType:
-        return (m_entry.doubleVal == other.m_entry.doubleVal) || (isnan(m_entry.doubleVal) && isnan(other.m_entry.doubleVal));
+        return (m_entry.doubleVal == other.m_entry.doubleVal) || (std::isnan(m_entry.doubleVal) && std::isnan(other.m_entry.doubleVal));
     case Options::Type::int32Type:
         return m_entry.int32Val == other.m_entry.int32Val;
