diff --git a/tools/llvm-config/CMakeLists.txt b/tools/llvm-config/CMakeLists.txt
index f59402ac4b0..70522aa20ed 100644
--- a/tools/llvm-config/CMakeLists.txt
+++ b/tools/llvm-config/CMakeLists.txt
@@ -11,6 +11,7 @@ add_llvm_tool(llvm-config
 # Compute the substitution values for various items.
 get_property(SUPPORT_SYSTEM_LIBS TARGET LLVMSupport PROPERTY LLVM_SYSTEM_LIBS)
 get_property(WINDOWSMANIFEST_SYSTEM_LIBS TARGET LLVMWindowsManifest PROPERTY LLVM_SYSTEM_LIBS)
+set_property(GLOBAL PROPERTY FIND_LIBRARY_USE_LIB64_PATHS 0)
 foreach(l ${SUPPORT_SYSTEM_LIBS} ${WINDOWSMANIFEST_SYSTEM_LIBS})
   if(MSVC)
     set(SYSTEM_LIBS ${SYSTEM_LIBS} "${l}.lib")
@@ -20,7 +21,13 @@ foreach(l ${SUPPORT_SYSTEM_LIBS} ${WINDOWSMANIFEST_SYSTEM_LIBS})
       set(SYSTEM_LIBS ${SYSTEM_LIBS} "${l}")
     else()
       # Otherwise assume it's a library name we need to link with.
-      set(SYSTEM_LIBS ${SYSTEM_LIBS} "-l${l}")
+      message(STATUS "LIB: ${l}")
+      find_library(ABSOLUTE_LIB "${l}")
+      message(STATUS "ABSOLUTE_LIB: ${ABSOLUTE_LIB}")
+      get_filename_component(LIBDIR "${ABSOLUTE_LIB}" DIRECTORY)
+      message(STATUS "LIBDIR: ${LIBDIR}")
+      set(SYSTEM_LIBS ${SYSTEM_LIBS} "-L${LIBDIR} -l${l}")
+      unset(ABSOLUTE_LIB CACHE)
     endif()
   endif()
 endforeach()
