diff --git a/tools/llvm-config/CMakeLists.txt b/tools/llvm-config/CMakeLists.txt
index 16ba54c0..79939955 100644
--- a/tools/llvm-config/CMakeLists.txt
+++ b/tools/llvm-config/CMakeLists.txt
@@ -11,6 +11,7 @@ add_llvm_tool(llvm-config
 # Compute the substitution values for various items.
 get_property(SUPPORT_SYSTEM_LIBS TARGET LLVMSupport PROPERTY LLVM_SYSTEM_LIBS)
 get_property(WINDOWSMANIFEST_SYSTEM_LIBS TARGET LLVMWindowsManifest PROPERTY LLVM_SYSTEM_LIBS)
+set_property(GLOBAL PROPERTY FIND_LIBRARY_USE_LIB64_PATHS 0)
 foreach(l ${SUPPORT_SYSTEM_LIBS} ${WINDOWSMANIFEST_SYSTEM_LIBS})
   if(MSVC)
     set(SYSTEM_LIBS ${SYSTEM_LIBS} "${l}.lib")
@@ -23,7 +24,10 @@ foreach(l ${SUPPORT_SYSTEM_LIBS} ${WINDOWSMANIFEST_SYSTEM_LIBS})
       if(IS_ABSOLUTE ${l})
         set(SYSTEM_LIBS ${SYSTEM_LIBS} "${l}")
       else()
-        set(SYSTEM_LIBS ${SYSTEM_LIBS} "-l${l}")
+        find_library(ABSOLUTE_LIB "${l}")
+        get_filename_component(LIBDIR "${ABSOLUTE_LIB}" DIRECTORY)
+        set(SYSTEM_LIBS ${SYSTEM_LIBS} "-L${LIBDIR} -l${l}")
+        unset(ABSOLUTE_LIB LIBDIR CACHE)
       endif()
     endif()
   endif()
